### PR TITLE
Date and time / it-academy.by / M-JC1-66-20 / Oxana Peregud / Done all tasks

### PR DESCRIPTION
**Pull request by OxanaPeregud**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        return localDate.plusMonths(months);
+    }
+
+    public static String localDatePattern(LocalDate localDate, String pattern) {
+        return localDate.format(DateTimeFormatter.ofPattern(pattern, US));
+    }
+
+    public static String localDateFormat(String localDate, String pattern) {
+        return LocalDate.parse(localDate).format(DateTimeFormatter.ofPattern(pattern, US));
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java:31]: Specification violation
```suggestion
    public static LocalDate localDateFormat(String localDate, String pattern) {
        return LocalDate.parse(localDate, DateTimeFormatter.ofPattern(pattern, US));
    }
```

File: src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        return temporal -> temporal.plus(Period.ofDays(days));
+    }
+
+    public static TemporalAdjuster firstJanuary(LocalDate localDate) {
+        int months = localDate.getMonthValue();
+        if (months < 6)
+            return temporal -> temporal.with(TemporalAdjusters.firstDayOfYear());
+        else
+            return temporal -> temporal.with(TemporalAdjusters.firstDayOfNextYear());
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java:47]: Wrong solution, example: 01.07.2020 -> 01.01.2020

**Review by rakovets — CHANGES_REQUESTED** [review]
Good job, but you need to fix the following problems

File: src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java (hunk 2 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/java/core/practice/date_and_time/DateWrapper.java:47]: Nothing wrong with the solution, it works fine. Added demo

**Review by rakovets — APPROVED** [review]
Approved.

**Pull request closed without merging**